js/ui/codingManager.js

```diff
@@ -29,239 +46,525 @@ function _isBuilderSpeedwagon(window) {
-                                  reactive: true,
-                                  can_focus: true,
-                                  x_fill: true,
-                                  y_fill: false,
-                                  track_hover: true });
-        let iconFile = Gio.File.new_for_uri('resource:///org/gnome/shell/theme/rotate.svg');
-        let gicon = new Gio.FileIcon({ file: iconFile });
-        let icon = new St.Icon({ style_class: 'view-source-icon',
-                                 gicon: gicon });
-        button.set_child(icon);
-        let rect = window.get_frame_rect();
-        button.set_position(rect.x + rect.width - BUTTON_OFFSET_X, rect.y + rect.height - BUTTON_OFFSET_Y);
-        Main.layoutManager.addChrome(button);
-        return button;
-    },
+    // Maybe admit this actor if it is the kind of actor that we want
+    admitBuilderWindowActor: function(actor) {
+        // If there is a currently bound window and it is not a speedwagon,
+        // then we can't admit this window. Return false.
+        if (this.builder &&
+            !Shell.WindowTracker.is_speedwagon_window(this.builder.meta_window))
+            return false;
 
-    _addSwitcherToApp: function(actorBuilder, session) {
-        session.actorBuilder = actorBuilder;
+        // We can admit this window. Wire up signals and synchronize
+        // geometries now.
+        this.builder = actor;
+        this.button.builder_window = actor.meta_window;
+
+        // The assumption here is that if we connect a new window, we
+        // are connecting a builder window (potentially 'on top') of the
+        // speedwagon window, so there is no need to disconnect
+        // signals
+        this._positionChangedIdBuilder = this.builder.meta_window.connect(
+            'position-changed', Lang.bind(this, this._synchronizeWindows)
+        );
+        this._sizeChangedIdBuilder = this.builder.meta_window.connect(
+            'size-changed', Lang.bind(this, this._synchronizeWindows)
+        );
+        _synchronizeMetaWindowActorGeometries(this.app, this.builder);
+
+        if (Shell.WindowTracker.is_speedwagon_window(actor.meta_window)) {
+            // If we are animating to a speedwagon window, we'll want to
+            // remove the 'above' attribute - we don't want the splash to
+            // appear over everything else.
+            actor.meta_window.unmake_above();
+        } else {
+            // We only want to untrack the coding app window at this
+            // point and not at the point we show the speedwagon. This
+            // will ensure that the shell window tracker is still
+            // watching for the builder window to appear.
+            this._stopWatchingForBuilderWindowToComeOnline();
 
-        this._animateToBuilder(session);
-    },
+            // We also only want to hide the speedwagon window at this
+            // point, since the other window has arrived.
+            this.splash.rampOut();
```

Comment:
In other words, does the splash instance stay around if you later receive another builder window? Or do you force a speedwagon every time the button is clicked, regardless of whether Builder is already running?